public/js/sidebar.js

```diff
@@ -1,47 +1,86 @@
 /* Handles the creation of the buttons to spawn towers
     and creeps */
 
-function createTowerButton (name, tid) {
-    var b = $('<input type="button" value="' 
-        + name + '" class="sideButton"'
-        + 'tid="' + tid + '"/>');
-    b.appendTo($("#towerButtons"));
-    b.click(function() {
-        this.tid = $(this).attr("tid")
-        var lastClick = myGrid.getLastClick();
-        alert("Tower Please: " + this.tid + "  " + lastClick.col + "  " + lastClick.row);
-        ws.towerRequest(this.tid, lastClick, "Tower please");
+var towerButtons = null, creepButtons = null;
```

Comment:
Instead of null here, it would be easier to read if this was empty array (i.e. var towerButtons = [], creepButtons = [];